File: palindromestring.java
import java.util.*;
import java.io.*;
class palindromestring
{
public static void main(String args[])
{
Scanner in=new Scanner(System.in);
System.out.println("Enter string");
String m1=in.nextLine();
int d=m1.length();
char ct[]=new char[20];
char st[]=new char[20];
for(int i=0;i<d;i++)
{
	ct[i]=m1.charAt(i);
}
for(int i=d-1;i>=0;i--)
{
      st[d-i-1]=ct[i];
   }
    System.out.println(ct);
     System.out.println(st);
   if(String.valueOf(st).equals(String.valueOf(ct)))
   {
   System.out.println("string is palindrome ");
   }
      else
      {
      System.out.println("string is not palindrome ");
   }
     }
 }

File: palindromestring.c
#include<stdio.h>
int main()
{
	char ar[100],rev[100];
	int i,n,flag;
	printf("enter array ");
	scanf("%[^\n]%*c",&ar);
	n=0;
	for(i=0;ar[i]!='\0';i++)
	{
		n++;
	}
	for(i=n-1;i>=0;i--)
	{
		rev[n-i-1]=ar[i];
}
for(i=0;i<n;i++)
{
		flag=0;
		if(ar[i]==rev[i])
		{
			flag=1;
		}
		else 
		{
			flag=0;
			break;
		}
	}
	if(flag==1)
	{
	printf("%s is a palindrome",ar);
}
	else
	{
		printf("%s is not a palindrome",ar);
	}
}

